mesh/src/common/AbstractMesh.cpp

```diff
@@ -509,6 +509,12 @@ void AbstractMesh<ELEMENT_DIM, SPACE_DIM>::SetMeshHasChangedSinceLoading()
     mMeshFileBaseName = "";
 }
 
+template<unsigned ELEMENT_DIM, unsigned SPACE_DIM>
```

Comment:
We definitely need this new method do we (seems like something we must be doing all over the place, probably with a `AbstractMesh::NodeIterator` instead?)